### PR TITLE
Move formatter<std::error_code> from fmt/os.h to fmt/std.h

### PR DESCRIPTION
**Pull request by phprus**
Issue: #3123

### CODE REVIEW

File: test/std-test.cc [hunk 1]
```diff
@@ -10,6 +10,7 @@
 #include <string>
 #include <vector>
 
+#include "fmt/os.h"  // fmt::system_category()
```
Comment (vitaut) [inline test/std-test.cc:13]: nit: we normally put symbol names without parentheses in comments like this

Comment (phprus) [reply test/std-test.cc]: Fixed.

**Comment by vitaut** [comment]
Thank you!

**Comment by tupaschoal** [comment]
Should this PR also have touched documentation for `std.h`? It currently does not say that it provides formatting for `error_code`, just:
```
fmt/std.h provides formatters for:

[std::filesystem::path](https://en.cppreference.com/w/cpp/filesystem/path)

[std::thread::id](https://en.cppreference.com/w/cpp/thread/thread/id)

[std::monostate](https://en.cppreference.com/w/cpp/utility/variant/monostate)

[std::variant](https://en.cppreference.com/w/cpp/utility/variant/variant)

[std::optional](https://en.cppreference.com/w/cpp/utility/optional)
```

**Pull request merged**